File: src/test/java/net/syscon/elite/service/impl/QueryParamHelperTest.java
package net.syscon.elite.service.impl;

import org.assertj.core.data.MapEntry;
import org.junit.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class QueryParamHelperTest {
    @Test
    public void splitTypes_empty() {
        final var map = QueryParamHelper.splitTypes(List.of());
        assertThat(map).isEmpty();
    }

    @Test
    public void splitTypes() {
        final var map = QueryParamHelper.splitTypes(List.of("BOB+JOE", "BOB+FRED", "HARRY", "JOHN+SMITH"));
        assertThat(map).containsKeys("BOB", "HARRY", "JOHN").containsExactly(
                MapEntry.entry("BOB", List.of("JOE", "FRED")),
                MapEntry.entry("HARRY", List.of()),
                MapEntry.entry("JOHN", List.of("SMITH")));
    }
}

File: src/test/java/uk/gov/justice/hmpps/prison/service/QueryParamHelperTest.kt
package uk.gov.justice.hmpps.prison.service

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.data.MapEntry.entry
import org.junit.jupiter.api.Test

class QueryParamHelperTest {
  @Test
  fun splitTypes_empty() {
    val map = QueryParamHelper.splitTypes(mutableListOf<String>())
    assertThat(map).isEmpty()
  }

  @Test
  fun splitTypes() {
    val map = QueryParamHelper.splitTypes(mutableListOf("BOB+JOE", "BOB+FRED", "HARRY", "JOHN+SMITH"))
    assertThat(map).containsKeys("BOB", "HARRY", "JOHN").containsExactly(
      entry("BOB", mutableListOf("JOE", "FRED")),
      entry("HARRY", mutableListOf()),
      entry("JOHN", mutableListOf("SMITH")),
    )
  }
}
